Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -313,7 +320,10 @@ class KalmanFitter {
              state.navigation.navigationBreak)) {
           // Remove the missing surfaces that occur after the last measurement
           result.missedActiveSurfaces.resize(result.measurementHoles);
-          if (reversedFiltering) {
+          // now get track state proxy for the smoothing logic
+          auto trackStateProxy =
+              result.fittedStates.getTrackState(result.lastTrackIndex);
```

Comment:
I guess this should be `lastMeasurementIndex`?

Reply:
Yes you are right thanks for catching that !